### ISSUE TITLE
--fuzz of HMAC formats leaks a lot of memory

### ISSUE THREAD

**Issue by solardiz** [openwall/john#4990]
This is not necessarily a bug - sometimes we accept continuing to allocate memory per each new input, and fuzzing produces a lot of inputs. This is more of an observation.

`--fuzz` of HMAC-MD5, HMAC-SHA224, HMAC-SHA256 leaks a lot of memory (tens of MB per second, and at least for HMAC-SHA256 eventually tens of GB total).

However, somehow HMAC-SHA1 and HMAC-SHA512 are mostly immune from this (they do leak memory, but much slower).

I don't know where this leak is. Possibly it comes from the allocations that `base64_convert()` can sometimes make.

**Comment by solardiz** [comment]
Looks like we won't care to get this fixed any time soon, if there's anything to fix here at all. Closing.

**Comment by AlekseyCherepanov** [comment]
I checked code of `base64_convert`. It does not seem there is a leak. (There is one in `base64->base64` conversion that is not relevant to `hmac-md5`.)

I tried to run `john` with some `hmac-md5` hashes populated with `--fuzz-dump=1,5000`. `Loaded 3004 password hashes with 2894 different salts`. It consumed ~1.7 Gb of memory. And ASan did not report any leaks.

`SALT_SIZE` is 294916 bytes for `hmac-md5`. Even so 3k allocated salts would be 843 Mb.

Fuzzer could reset the database periodically to reduce memory usage.
